broker/pkg/coordinator/grpcserver.go

```diff
@@ -244,11 +245,14 @@ func (s *GRPCServer) TimelineDelete(stream grpc.Broker_TimelineDeleteServer) err
 			}
 			return err
 		}
+		consumer, err := s.listeners.GetConsumer(stream.Context(), string(req.SessionID()))
+		if err != nil {
+			return err
+		}
 
 		batch = append(batch, timeline.Message{
 			ID:              req.MessageIDBytes(),
-			ProducerGroupID: nil,
-			LockConsumerID:  nil,
+			LockConsumerID:  consumer.ID,
```

Comment:
You cannot make this assumption at this level, any other consumer can make this call, is the Storage responsability to not allow it if the caller does not have the right to remove it.